Massive rework for .18

TimVroom opened the pull request:
Reworked to be up to coding standards for 0.18 / 1.0
Currently incorrect/now working: train schedules/trips.

Added a lot of stats to the exporting (force/robots etc)
Added power for exporting
Added mod/game info to the export.

TimVroom committed:
Remastering of graftorio for nilaus' usage

TimVroom committed:
Minor fixes...

TimVroom committed:
Minor train fix

TimVroom committed:
Adding additional and multiple stats to track

TimVroom committed:
Merge branch 'master' of github.com:afex/graftorio

TimVroom committed:
Minor typo's

TimVroom committed:
Bugfixes and more cleanup.

- More localized vars
- Proper export of values such as research queue
- Removed some unneeded events in loading

TimVroom committed:
Massive change - separated to different libs so its more manageable

TimVroom committed:
Changed one missed game.tick to event.tick

TimVroom committed:
Updated version for deploy

TimVroom committed:
Added power tracker

TimVroom committed:
Updated version and seperate calculation before sending out in different
ticks

TimVroom committed:
Performance improvements

Disabled train tracking for now (instable)
Added polution tracking
Added total charging bots for every cell in a network

TimVroom committed:
Updated it all again...

TimVroom committed:
Minor fix for deconstruction

TimVroom committed:
Updated version

TimVroom committed:
Added example dashboards
Updated readme.
Added example env.grafana

TimVroom committed:
Added plugin support so other mods can hook into graftorio

TimVroom committed:
Made sure events are not in global space, they can change when altering
mods...
Updated version number

TimVroom committed:
Updated version and changelog.

Make sure some gauges are reset when needed

TimVroom committed:
Make sure its the same versioning

TimVroom committed:
Make sure plugins can actually work...

Removed some unuseable prometheus stuff

TimVroom committed:
Updated version

TimVroom committed:
Updated to .15, making sure plugins works

TimVroom committed:
Added automatic translation using the players. SHould only be done once
the game is loaded.

TimVroom committed:
Updated so it keeps original full name

TimVroom committed:
Fixed typo

TimVroom committed:
Updated translations and fixing research queue (mostly)

TimVroom committed:
Better fix for correct level selection

SnO2WMaN commented:
@TimVroom Your rework version works very well (at least in my env). However this original mod owners seems not to be active on GitHub for a long time, so I think this pull request may not to be merged. Do you have a thought to release your reworked version as like *graftorio2*?

P.S. I found some typo in README.md and mod output data.

TimVroom committed:
Updated to .20

TimVroom committed:
Added distribute code.
Updated so it allows for row titles to include icons

added # option to keep - instead of replacing with _

attrib commented:
Also tested it quickly and all seems to work with 1.0. But I do not want to enable the whole time, as this would need all users to download it manually. So can't test it currently with multiple players online at the moment.

TimVroom committed:
Removed proxies, but replaced all localised strings with their
item/entity prototype when needed.

TimVroom committed:
Minor fixes and more info for export

TimVroom committed:
Updated changelog

TimVroom committed:
Updated translation to not be crashing...

TimVroom committed:
Minor fix again...

TimVroom committed:
Split out the events more so it would clutter less.

Make sure the export function isn't to big and causes lag spikes.
Chunked the collectors up and do it in a few ticks instead of 1.

Disabled the logistics collecting again (too intensive for big networks)

TimVroom committed:
Forgot control.lua

TimVroom committed:
Updated version

TimVroom committed:
Updated tick delay and made server only save possible

TimVroom committed:
Merge branch 'development' into 'master'

Updated tick delay and made server only save possible

See merge request thevirtualcrewprojects/graftorio!1

wbourne0 commented:
The mod does seem to work with v1.0 but we're getting this issue occasionally when saving the map:
```
2427.751 Error ParallelScenarioSaver.cpp:114: Saving scenario failed: The mod graftorio (1.0.25) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event graftorio::on_save()
LuaCustomTable cannot be serialized.
2427.761 Error Util.cpp:83: Attempting to create notice box in headless mode. Message: 'Cannot save map: The mod graftorio (1.0.25) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event graftorio::on_save()
LuaCustomTable cannot be serialized.'
```

TimVroom committed:
Make sure to not crash while a save gets mods updates while being in
export mode

TimVroom committed:
Additional encoding layer because it still crashed.

Also it didn't clear the translated stuff when a mod changed

wbourne0 commented:
@TimVroom do these fix the issue that I pointed out?

TimVroom commented:
> Also tested it quickly and all seems to work with 1.0. But I do not want to enable the whole time, as this would need all users to download it manually. So can't test it currently with multiple players online at the moment.

Yes it seems to work okay ish on 1.0. There is currently a setting that is enabled by default to only save on the server, so that saves a little bit of ups for the users.

> @TimVroom do these fix the issue that I pointed out?

It should be yes, be aware that it might be a bit slower on bigger bases.

jlucktay commented:
I recommend taking a look at the new `graftorio2` fork [here](https://github.com/remijouannet/graftorio2) and recreating/adapting this PR as necessary. 😃
